hs_geo_raster_resource/receivers.py

```diff
@@ -1,48 +1,172 @@
 ## at the end of the models.py for the import of this module
+import tempfile
+import os
+import subprocess
+import shutil
+import zipfile
+from collections import OrderedDict
+import xml.etree.ElementTree as ET
+
+import gdal
+from gdalconst import GA_ReadOnly
+
+from django.core.files.uploadedfile import UploadedFile
 from django.dispatch import receiver
-from hs_core.signals import *
-from forms import *
+
+from hs_core.hydroshare.resource import ResourceFile
+from hs_core.signals import pre_create_resource, pre_add_files_to_resource, pre_delete_file_from_resource, \
+    pre_metadata_element_create, pre_metadata_element_update
+from forms import CellInfoValidationForm, BandInfoValidationForm, OriginalCoverageSpatialForm
+from models import RasterResource
+import raster_meta_extract
 
 # signal handler to extract metadata from uploaded geotiff file and return template contexts
 # to populate create-resource.html template page
 
 
+def raster_file_validation(files):
+    is_valid = True
+    vrt_file_path = ''
+
+    # process uploaded .tif or .zip file
+    if len(files) == 1:
+        ext = os.path.splitext(files[0].name)[1]
+        if ext == '.tif':
+            temp_vrt_file_path, temp_dir = create_vrt_file(files[0])
+            if os.path.isfile(temp_vrt_file_path):
+                files.append(UploadedFile(file=open(temp_vrt_file_path, 'r'), name=os.path.basename(temp_vrt_file_path)))
+
+        elif ext == '.zip':
+            extract_file_paths, temp_dir = explode_zip_file(files[0])
+            if extract_file_paths:
+                del files[0]
+                for file_path in extract_file_paths:
+                    files.append(UploadedFile(file=open(file_path, 'r'), name=os.path.basename(file_path)))
+
+    # check if raster is valid in format and data
+    files_names = [f.name for f in files]
+    files_ext = [os.path.splitext(path)[1] for path in files_names]
+    files_path = [f.file.name for f in files]
+
+    if set(files_ext) == {'.vrt', '.tif'} and files_ext.count('.vrt') == 1:
+        vrt_file_path = files_path[files_ext.index('.vrt')]
+        raster_dataset = gdal.Open(vrt_file_path, GA_ReadOnly)
+
+        # check if the data is valid
+        try:
+            raster_dataset.RasterXSize
+            raster_dataset.RasterYSize
+            raster_dataset.RasterCount
+        except AttributeError:
+            is_valid = False
+
+        # check if the raster file numbers and names are valid in vrt file
+        with open(vrt_file_path, 'r') as vrt_file:
+            vrt_string = vrt_file.read()
+            root = ET.fromstring(vrt_string)
+            raster_file_names = {file_name.text for file_name in root.iter('SourceFilename')}
+        files_names.pop(files_ext.index('.vrt'))
+        if raster_file_names != set(files_names):
+            is_valid = False
+    else:
+        is_valid = False
+
+    return is_valid, vrt_file_path, temp_dir
```

Comment:
It seems there is no command to directly validate if a vrt is valid or not. It may show some error info if the vrt raster is manipulated or processed by some of the commands when the file is not valid.